svm/src/account_loader.rs

```diff
@@ -208,30 +208,42 @@ fn load_transaction_accounts<CB: TransactionProcessingCallback>(
         .unique()
         .collect::<Vec<&u8>>();
 
+    let mut fee_payer_account = Some((
+        tx_details.fee_payer_account.data().len(),
+        tx_details.fee_payer_account,
+        tx_details.fee_payer_rent_debit,
+    ));
+
     let mut accounts = account_keys
         .iter()
         .enumerate()
         .map(|(i, key)| {
             let mut account_found = true;
-            #[allow(clippy::collapsible_else_if)]
-            let account = if solana_sdk::sysvar::instructions::check_id(key) {
-                construct_instructions_account(message)
-            } else {
-                let is_fee_payer = i == 0;
-                let instruction_account = u8::try_from(i)
-                    .map(|i| instruction_accounts.contains(&&i))
-                    .unwrap_or(false);
-                let (account_size, account, rent) = if is_fee_payer {
+            let is_instruction_account = u8::try_from(i)
+                .map(|i| instruction_accounts.contains(&&i))
+                .unwrap_or(false);
+            let (account_size, account, rent) =
+                if let Some(fee_payer_account) = fee_payer_account.take() {
```

Comment:
For me, this logic is harder to follow. I'd rather have this separated entirely from the loop than using this option taking tbh

Reply:
Ok, I personally found this more readable but did a refactor here: https://github.com/anza-xyz/agave/pull/2442/commits/f38ba7a55b066283da869bffd9ed1f263545bcd7, let me know what you think